### PR TITLE
Show the federated IdP in the CLI instead of the Sigstore DEX instance

### LINKED ISSUE #157: Fix IdP shown in CLI
See https://github.com/sigstore/sigstore-python/issues/970

We need to extract data from the cert ourselves to get the IdP shown to users, instead of using the Sigstore DEX instance.

Update should be this line https://github.com/sigstore/model-transparency/blob/main/model_signing/model.py#L104

### PR DESCRIPTION
**Pull request by mayaCostantini**
Resolves #157

#### Summary

Show the federated IdP instead of the Sigstore DEX instance, extracted from `sigstore.oidc.IdentityToken`.
As mentioned in the comment, this is more of a workaround for versions of `sigstore-python` earlier than `3.0.0`, as the `federated_issuer` property could be called directly in the latest version.

#### Release Note

Replaced DEX instance with federated IdP in signing CLI field `identity-provider`.

#### Documentation

No documentation required.

### CODE REVIEW

**Review by mihaimaruseac — DISMISSED** [review]
Thank you!

**Comment by mihaimaruseac** [comment]
Oh, can you fix the lint issues, please (lines too long)

File: model_signing/model.py [hunk 1]
```diff
@@ -101,7 +101,10 @@ def sign(self, inputfn: Path, signaturefn: Path,
             oidc_token = self.get_identity_token()
             if not oidc_token:
                 raise ValueError("No identity token supplied or detected!")
-            print(f"identity-provider: {oidc_token.issuer}",
+            # Calling the private attribute IdentityToken._federated issuer is a workaround
+            # For earlier versions of sigstore-python (<3.0.0) that do not support the
+            # federated_issuer property.
+            print(f"identity-provider: {oidc_token._federated_issuer}",
```
Comment (haydentherapper) [inline model_signing/model.py:107]: Is federated issuer always set? If a non-Dex token is used, what is this value?

Comment (laurentsimon) [reply model_signing/model.py:107]: @woodruffw please advise

Comment (laurentsimon) [reply model_signing/model.py:107]: @mayaCostantini this is the original issue + solution that @woodruffw told us should work https://github.com/sigstore/sigstore-python/issues/970#issuecomment-2108727868. So the right solution is probably `.expected_certificate_subject`

**Review by laurentsimon — COMMENTED** [review]
Thanks for the PR!

File: model_signing/model.py (hunk 1 above)
Comment (woodruffw) [reply model_signing/model.py:107]: With `sigstore-python` 3.0.0, `federated_issuer` should fall back to the "top-level" issuer if the Dex-specific federation claim isn't present. You can see that here:

https://github.com/sigstore/sigstore-python/blob/29c98ea7f58c3047b9b5e062af912f4523ff1514/sigstore/oidc.py#L208-L227

(Before 3.0.0, `federated_issuer` was called `expected_certificate_subject`, which was a misleading name. But they're the same API.)

**Comment by haydentherapper** [comment]
Can you fix the lint failure? LGTM otherwise, thanks!

**Comment by mayaCostantini** [comment]
Thanks for the reviews, the linting issues should be fixed now.

**Review by mihaimaruseac — APPROVED** [review]

**Comment by mihaimaruseac** [comment]
Thank you!

**Pull request merged**